File: go/p6.go
package main

import "fmt"
import "math"
import "os"
import "strconv"

func main() {
  var limit int = 100
  if len(os.Args) > 1 {
    if s, err := strconv.Atoi(os.Args[1]); err == nil {
      limit = int(s)
    }
  }
  var sum int = limit * (limit + 1) / 2
  var sum_sq int = (2 * limit + 1) * (limit + 1) * limit / 6
  var diff int = int(math.Pow(float64(sum), 2.0)) - sum_sq
  fmt.Printf("%v\n", diff)
}

File: r/p6.r
#!/usr/bin/Rscript

printf <- function(...) invisible(print(sprintf(...)))

limit  <- 100
args  <- commandArgs(TRUE)
if(length(args) > 0){
  limit  <- as.integer(args[1])
}
sum  <- limit * (limit + 1) / 2
sumsq <- (2 * limit + 1) * (limit + 1) * limit / 6
diff  <- '^'(sum, 2) - sumsq

printf("%i", diff)
